### PR TITLE
Various fixes found during interop testing

### PR DESCRIPTION
**Pull request by glbrntt**
Interoperability testing found a number of small bugs:
- We didn't handle "Trailers-Only" responses
- We didn't handle non-ASCII status messages correctly
- The timeout passed to the server incorrectly had an space in it
- Trailing metadata was never passed down the pipeline to the caller

### CODE REVIEW

File: Sources/SwiftGRPCNIO/GRPCHeaderName.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+
+public struct GRPCHeaderName {
```
Comment (kevints) [inline Sources/SwiftGRPCNIO/GRPCHeaderName.swift]: Since this is never instantiated consider an empty enum instead

Comment (MrMage) [reply Sources/SwiftGRPCNIO/GRPCHeaderName.swift]: For what it's worth I'm always torn between an empty enum and a struct with a private initializer.

Comment (glbrntt) [reply Sources/SwiftGRPCNIO/GRPCHeaderName.swift]: Other than not being able to instantiate the `enum` is there any difference?

**Comment by glbrntt** [comment]
> LGTM. Should we have extra tests for e.g. passing the trailing metadata and/or non-ASCII status messages, or are these covered sufficiently by the interop tests?

I think we're covered sufficiently by the interop tests.

File: Sources/SwiftGRPCNIO/GRPCHeaderName.swift (hunk 1 above)
Comment (MrMage) [reply Sources/SwiftGRPCNIO/GRPCHeaderName.swift]: I don't think so. Maybe @kevints knows more?

**Comment by glbrntt** [comment]
@MrMage can this be merged?

**Comment by MrMage** [comment]
Yes, sorry. I had just been waiting for a comment from @kevints, but I think this is good to go as-is.

**Pull request merged**